doc/man7/provider-base.pod

```diff
@@ -70,6 +70,12 @@ provider-base
  int BIO_vprintf(OSSL_CORE_BIO *bio, const char *format, va_list args);
  int BIO_vsnprintf(char *buf, size_t n, const char *fmt, va_list args);
 
+ int OBJ_add_sigid(int signid, int dig_id, int pkey_id);
+ int OBJ_create(const char *oid, const char *sn, const char *ln);
+ int OBJ_txt2nid(const char *s);
+ int OBJ_ln2nid(const char *s);
+ int OBJ_sn2nid(const char *s);
```

Comment:
With txt2nid, I really don't see the necessity for sn2nid and ln2nid

Reply:
I added them for the sake of completeness. I can remove them if that is the consensus opinion.

Reply:
Will we be able to get rid of support for nids if we set it in stone like this? Basically a provider working with OpenSSL 3.0 would not be able to work with a fictional future OpenSSL 5.0 release which does not use nids anymore. And/or should we add provctx parameter to these callbacks that would not be used for now but could be used in future if we implemented something like per-libctx OBJECT database?

Reply:
> I added them for the sake of completeness. I can remove them if that is the consensus opinion.

They should be kept as a block IMHO - so "completeness" makes sense.

We depend on such a mapping existing - which is why it is being added - and until we introduce another mechanism this is what has to be used. When a new mechanism is put in place we can figure out what to do for this and what approach to take in terms of forward and backward compatibility.

This is a pragmatic solution to a current problem without entangling in a completely new mechanism at a very late stage.

Reply:
I'm more thinking that in terms of algorithms, we see this all over the place:

``` C
    nid = OBJ_sn2nid(whatever);
    if (nid == NID_undef)
        nid = OBJ_ln2nid(whatever);
```

(except in some places, where OBJ_ln2nid() seems to be forgotten for no good reason)

In all those places, OBJ_txt2nid() could just as well be used, and should be recommended.  I can't see a spot in provider code where this isn't applicable.